dynamic_programming/LCS.c

```diff
@@ -0,0 +1,212 @@
+/**
+ * @file
+ * @author [Santosh Jonnakuti](https://github.com/Santoshjonnakuti)
+ * @brief [Longest Common Subsequence Algorithm](https://en.wikipedia.org/wiki/Longest_common_subsequence_problem)
+ * implementation in C to find the LCS between 2 strings.
+ * @details LCS algorithm uses dynamic approach to find all possible LCS between 2 strings. 
+ * Dynamic Programming follow a method so that even sub-problems will also have optimal solution.
+ *
+ * LCS() will find the length of the LCS of the following two strings 'GAC' and 'AGCAT'.
+ * LCS function will return the follwing 2D-Matrix
+ *
+ *	  0 0 0 0 0 0
+ *      0 0 1 1 1 1
+ *      0 1 1 1 2 2
+ *      0 1 1 2 2 2
+ *
+ * So Length of LCS will be 2.
+ *
+ * allPossibleLSC() will print all possible LCS between those strings.
+ *
+ * AC
+ * GC
+ * GA
+ *
+ * So There are 3 Longest common subsequences in 'GAC' and 'AGCAT'
+ */
+
+
+#include<stdio.h>     /// for I/O operations
+#include<stdlib.h>   ///  for Dynamic memory allocation
+#include<string.h>  ///   for puts(), gets() and strlen()
+
+#define MAX 100      // Declaration of a constant value
```

Comment:
Provide a better description of this constant variable.

Reply:
Updated all the Changes Requested Please review @Panquesito7  as soon as you can.
Thank You!